src/main/java/uk/gov/justice/digital/job/batchprocessing/ReloadDiffProcessor.java

```diff
@@ -0,0 +1,115 @@
+package uk.gov.justice.digital.job.batchprocessing;
+
+import jakarta.inject.Inject;
+import lombok.val;
+import org.apache.spark.sql.Column;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.expressions.Window;
+import org.apache.spark.sql.functions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.collection.JavaConverters;
+import scala.collection.Seq;
+import uk.gov.justice.digital.datahub.model.SourceReference;
+import uk.gov.justice.digital.service.DataStorageService;
+
+import javax.inject.Singleton;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+
+import static org.apache.spark.sql.functions.col;
+import static org.apache.spark.sql.functions.lit;
+import static org.apache.spark.sql.functions.not;
+import static org.apache.spark.sql.functions.row_number;
+import static uk.gov.justice.digital.common.CommonDataFields.CHECKPOINT_COL;
+import static uk.gov.justice.digital.common.CommonDataFields.OPERATION;
+import static uk.gov.justice.digital.common.CommonDataFields.ShortOperationCode.*;
+import static uk.gov.justice.digital.common.ResourcePath.createValidatedPath;
+
+@Singleton
+public class ReloadDiffProcessor {
+
+    private static final Logger logger = LoggerFactory.getLogger(ReloadDiffProcessor.class);
+    private static final String LEFT_ANTI_JOIN_TYPE = "leftanti";
+    private static final String RANK_COL = "rank";
+    private static final String DATE_TIME_PATTERN = "yyyyMMddHHmmss";
+    private final DataStorageService storageService;
+
+    @Inject
+    public ReloadDiffProcessor(DataStorageService storageService) {
+        this.storageService = storageService;
+    }
+
+
+    public void createDiff(SourceReference sourceReference, String outputBasePath, Dataset<Row> raw, Dataset<Row> archive, Date reloadTime) {
+        String formattedStartTime = new SimpleDateFormat(DATE_TIME_PATTERN).format(reloadTime);
+        val windowFunction = Window.partitionBy(getKeyColumnsAsSeq(sourceReference)).orderBy(col(CHECKPOINT_COL).desc());
+
+        Dataset<Row> activeArchiveRecords = archive
+                .withColumn(RANK_COL, row_number().over(windowFunction))
+                .where(col(RANK_COL).eqNullSafe(lit(1)).and(col(OPERATION).notEqual(lit(Delete.getName()))))
+                .drop(RANK_COL)
+                .persist();
```

Comment:
The diffs are computed against the active archive records. These are the most recent records for each primary key. Deletes are discarded because as they are no longer active.